pkg/shares/non_interactive_defaults.go

```diff
@@ -83,10 +83,27 @@ func roundUpBy(cursor, v int) int {
 	}
 }
 
-// MinSquareSize returns the minimum square size that can contain shareCount
-// number of shares.
-func MinSquareSize(shareCount int) int {
-	return RoundUpPowerOfTwo(int(math.Ceil(math.Sqrt(float64(shareCount)))))
+// SubTreeSize determines how many shares should be included when creating subtree. This is used for determining the padding and the reasoning behind this
+// algorithm is discussed in depth in ADR013.
+func SubTreeSize(shareCount int) int {
+	// per ADR013, we use a predetermined threshold to determine the padding
+	// needed
+	s := (shareCount / appconsts.SubtreeRootHeightThreshold)
+
+	// round up if the batch size is not an exact multiple of the threshold
+	if shareCount%appconsts.SubtreeRootHeightThreshold != 0 {
+		s++
+	}
+
+	// use the next power of two, as subtree roots inherently require this
+	s = RoundUpPowerOfTwo(s)
+
+	// per ADR013, use the min square size should that ever raise
+	if s < appconsts.DefaultMinSquareSize {
+		return appconsts.DefaultMinSquareSize
+	}
+
+	return s
```

Comment:
This is the only necessary meaningful logic change!!! the rest is just fixing tests lol